src/azure-cli-core/azure/cli/core/keys.py

```diff
@@ -32,14 +35,33 @@ def is_valid_ssh_rsa_public_key(openssh_pubkey):
 def generate_ssh_keys(private_key_filepath, public_key_filepath):
     import paramiko
 
+    if os.path.exists(public_key_filepath):
+        with open(public_key_filepath, 'r') as public_key_file:
+            public_key = public_key_file.read()
+
+            pub_ssh_dir, _ = os.path.split(public_key_filepath)
+            logger.warning("Public SSH key file '%s' found in dir '%s'."
+                           "Use public key file. New RSA key pair will not be generated.",
+                           public_key_filepath, pub_ssh_dir)
+
+            return public_key
```

Comment:
``` Python
try:
    with open(public_key_filepath, 'r') as public_key_file:
        public_key = public_key_file.read()
        pub_ssh_dir, _ = os.path.split(public_key_filepath)
        logger.warning("Public SSH key file '%s' found in dir '%s'."
                       "Use public key file. New RSA key pair will not be generated.",
                       public_key_filepath, pub_ssh_dir)
        return public_key
except IOError:
    pass
```

Two reasons:
1. Pythonic principal [Easier to ask for forgiveness than permission](https://docs.python.org/3/glossary.html#term-eafp). [Explaination](https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python)
2. Also, capture the issue when the public key is unreadable.

Reply:
I will address your comments, @troydai. Somehow I missed your other comments.